app/lib/katello/validators/environment_docker_repositories_validator.rb

```diff
@@ -0,0 +1,67 @@
+module Katello
+  module Validators
+    class EnvironmentDockerRepositoriesValidator < ActiveModel::Validator
```

Comment:
something is wrong with this validator, it didn't seem to prevent duplicate repo-registry-ids

Reply:
Updated to check on publish and promote. More tests still to write.

Reply:
Composite CVs not handled, will fix

Reply:
All fixed, composite CVs handled properly.